apps/gitness/src/pages-v2/project/project-rules-create-container.tsx

```diff
@@ -0,0 +1,208 @@
+  RepoBranchSettingsRulesPage
+} from '@harnessio/ui/views'
+
+import { useGetSpaceURLParam } from '../../framework/hooks/useGetSpaceParam'
+import { useMFEContext } from '../../framework/hooks/useMFEContext'
+import { useTranslationStore } from '../../i18n/stores/i18n-store'
+import { transformDataFromApi, transformFormOutput } from '../../utils/repo-branch-rules-utils'
+import { useBranchRulesStore } from '../repo/stores/repo-branch-rules-store'
+import { useProjectRulesStore } from './stores/project-rules-store'
+
+export const ProjectRulesCreateOrUpdateContainer = () => {
+  const { t } = useTranslationStore()
+  const navigate = useNavigate()
+
+  const spaceURL = useGetSpaceURLParam()
+
+  const { ruleId: ruleIdentifier } = useParams()
+  const { setPresetRuleData, setPrincipals, setRecentStatusChecks } = useProjectRulesStore()
+  const [principalsSearchQuery, setPrincipalsSearchQuery] = useState('')
+  const { dispatch } = useBranchRulesStore()
+  const [isSubmitSuccess, setIsSubmitSuccess] = useState<boolean>()
+  const {
+    scope: { accountId }
+  } = useMFEContext()
+
+  const branchRules = useMemo(() => {
+    return getBranchRules(t)
+  }, [t])
+
+  /**
+   * Reset form data
+   */
+  useEffect(() => {
+    return () => {
+      setPresetRuleData(null)
+      setPrincipals(null)
+      setRecentStatusChecks(null)
+    }
+  }, [setPresetRuleData, setPrincipals, setRecentStatusChecks])
+
+  const { data: { body: rulesData } = {} } = useSpaceRuleGetQuery(
+    { space_ref: spaceURL ?? '', rule_identifier: ruleIdentifier ?? '' },
+    {
+      enabled: !!ruleIdentifier
+    }
+  )
+
+  const {
+    mutate: addRule,
+    error: addRuleError,
+    isLoading: addingRule
+  } = useSpaceRuleAddMutation(
+    { space_ref: spaceURL },
+    {
+      onSuccess: () => {
+        setIsSubmitSuccess(true)
+        navigate('..')
+      }
+    }
+  )
+
+  const { data: { body: principals } = {}, error: principalsError } = useListPrincipalsQuery({
+    // @ts-expect-error : BE issue - not implemnted
+    queryParams: { page: 1, limit: 100, type: 'user', query: principalsSearchQuery, accountIdentifier: accountId }
+  })
+
+  const {
+    mutate: updateRule,
+    error: updateRuleError,
+    isLoading: updatingRule
+  } = useSpaceRuleUpdateMutation(
+    { space_ref: spaceURL, rule_identifier: ruleIdentifier! },
+    {
+      onSuccess: () => {
+        setIsSubmitSuccess(true)
+        navigate('..')
+      }
+    }
+  )
+  const { data: { body: recentStatusChecks } = {}, error: statusChecksError } = useListStatusCheckRecentSpaceQuery({
+    space_ref: spaceURL ?? '',
+    queryParams: {}
+  })
+
+  const handleRuleUpdate = (data: RepoBranchSettingsFormFields) => {
+    const formattedData = transformFormOutput(data)
+
+    if (ruleIdentifier) {
+      // Update existing rule
+      updateRule({
+        body: formattedData
+      })
+    } else {
+      // Add new rule
+      addRule({
+        body: formattedData
+      })
+    }
```

Comment:
this can be potentially simplified for all use add/edit usecases:

```suggestion
    const addOrUpdate = ruleIdentifier ? updateRule : addRule
    addOrUpdate({
      body: formattedData
    })
```

Reply:
Since the two functions have different path_params, TS is not happy with this.